### PR TITLE
Create fake gRPC server

### PR DESCRIPTION
**Pull request by barcohen10**
https://app.zenhub.com/workspaces/ae-platform-team-5a41ffc34e0b5a51032a968a/issues/soluto-private/ae-platform/1328

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,3 @@
 # soluto-kafka
+
+add .npmrc file
```
Comment (Barmil) [inline README.md:3]: ?

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,7952 @@
+{
```
Comment (Barmil) [inline package-lock.json:1]: We use yarn, so you dont need to upload this file.

File: tests/FakeGRPC.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+WORKDIR /app
+
+COPY .npmrc .
+
+COPY fakeGRPCServer/package.json .
+COPY fakeGRPCServer/yarn.lock .
+
+RUN yarn
+
+COPY fakeGRPCServer .
```
Comment (Barmil) [inline tests/FakeGRPC.Dockerfile:12]: did you find out why do we copy that way?

File: tests/specs/tests.spec.ts [hunk 4]
```diff
@@ -151,6 +152,41 @@ describe('tests', () => {
...
 
+describe('gRPC tests', () => {
+    const FAKE_GRPC_SERVER_URL = `http://localhost:${PORTS.express}`;
+
+    beforeEach(async () => {
+        await fetch(`${FAKE_GRPC_SERVER_URL}/clear`);
+    });
+
+    it('should pass all data and headers', async () => {
+        const topic = 'bar';
```
Comment (Barmil) [inline tests/specs/tests.spec.ts:163]: good topic name

File: tests/fakeGRPCServer/package.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (Barmil) [inline tests/fakeGRPCServer/package.json:1]: You should consider splitting it into a package.

File: tests/FakeGRPC.Dockerfile [hunk 6]
```diff
@@ -0,0 +1,18 @@
+FROM node:14-slim
+
+WORKDIR /app
+
+COPY .npmrc .
```
Comment (Barmil) [inline tests/FakeGRPC.Dockerfile:5]: generally, this package considered open-source. so we need to consider if we want to do it.

File: tests/fakeGRPCServer/gRPCServer.ts [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import {startServer} from '@npmsoluto/soluto-kafka-grpc-target';
```
Comment (Barmil) [inline tests/fakeGRPCServer/gRPCServer.ts:1]: I don't think you should use that package here. (again open-source vs private package)

**Review by Barmil — CHANGES_REQUESTED** [review]

File: tests/FakeGRPC.Dockerfile (hunk 3 above)
Comment (barcohen10) [reply tests/FakeGRPC.Dockerfile:12]: @Barmil if we copy the entire codebase before installing our dependencies, on each change we make during development Docker will have to reinstall all our dependencies, even though most of the time they haven’t changed

**Pull request closed without merging**